azure/linux-virtual-machine/main.tf

```diff
@@ -181,3 +181,18 @@ resource "azurerm_monitor_data_collection_rule_association" "main" {
 
   depends_on = [azurerm_virtual_machine_extension.main_azmonitor]
 }
+
+resource "azurerm_dev_test_global_vm_shutdown_schedule" "main" {
+  count = var.autoshutdown != null ? 1 : 0
+
+  virtual_machine_id = azurerm_linux_virtual_machine.main.id
+  location           = var.location
+  tags               = var.tags
+
+  daily_recurrence_time = var.autoshutdown["time"]
```

Comment:
Not really a comment on this specific commit but might open a more general conversation about style guidelines.

I would have written:
`var.autoshutdown.time`

You have written:
`var.autoshutdown["time"]`

Both are valid and this code is fine to go in but it would make sense to decide on a standard between ourselves generally to make the code more consistent. Outside the scope of this PR but worth a note.

Reply:
Personally, I feel that it makes more sense to access a map with `[]` notation and structured objects with `.` notation. It allows the code reader to distinguish between them. The question is however is `_"should the reader need to distinguish between them?"_`

Reply:
At current this is what's used in a few modules, e.g. in the azuredevops-project module.